src/core/openthread.cpp

```diff
@@ -977,33 +979,33 @@ void HandleActiveScanResult(void *aContext, Mac::Frame *aFrame)
         memcpy(&result.mExtendedPanId, beacon->GetExtendedPanId(), sizeof(result.mExtendedPanId));
     }
 
-    handler(&result);
+    handler(&result, aClientContext);
 
 exit:
     return;
 }
 
 ThreadError otDiscover(uint32_t aScanChannels, uint16_t aScanDuration, uint16_t aPanId,
-                       otHandleActiveScanResult aCallback)
+                       otHandleActiveScanResult aCallback, void *aCallbackContext)
 {
     return sThreadNetif->GetMle().Discover(aScanChannels, aScanDuration, aPanId, &HandleMleDiscover,
-                                           reinterpret_cast<void *>(aCallback));
+                                           reinterpret_cast<void *>(aCallback), aCallbackContext);
```

Comment:
I understand that, but I think the additional state in the API layer could be rolled up with the introduction of `otContext` in #357.